libraries/node-core-library/src/test/Sort.test.ts

```diff
@@ -69,3 +69,192 @@ test('Sort.sortSet', () => {
   Sort.sortSet(set);
   expect(Array.from(set)).toEqual(['aardvark', 'goose', 'zebra']);
 });
+
+describe('Sort.sortKeys', () => {
+  // Test cases from https://github.com/sindresorhus/sort-keys/blob/v4.2.0/test.js
+  function deepEqualInOrder(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    actual: Partial<Record<string, any>>,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    expected: Partial<Record<string, any>>
+  ): void {
+    expect(actual).toEqual(expected);
+
+    const seen = new Set();
+
+    function assertSameKeysInOrder(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      object1: Partial<Record<string, any>>,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      object2: Partial<Record<string, any>>
+    ): void {
+      // This function assumes the objects given are already deep equal.
+
+      if (seen.has(object1) && seen.has(object2)) {
+        return;
+      }
+
+      seen.add(object1);
+      seen.add(object2);
+
+      if (Array.isArray(object1)) {
+        for (const index of object1.keys()) {
+          assertSameKeysInOrder(object1[index], object2[index]);
+        }
+      } else if (typeof object1 === 'object') {
+        const keys1 = Object.keys(object1);
+        const keys2 = Object.keys(object2);
+        expect(keys1).toEqual(keys2);
+        for (const index of keys1.keys()) {
+          assertSameKeysInOrder(object1[keys1[index]], object2[keys2[index]]);
+        }
+      }
+    }
+
+    assertSameKeysInOrder(actual, expected);
+  }
+
+  test('sort the keys of an object', () => {
+    const unsortedObj = { c: 0, a: 0, b: 0 };
+    const sortedObj = Sort.sortKeys(unsortedObj);
+    // Assert that it's not sorted in-place
+    expect(sortedObj).not.toBe(unsortedObj);
+    deepEqualInOrder(unsortedObj, { c: 0, a: 0, b: 0 });
+    deepEqualInOrder(sortedObj, { a: 0, b: 0, c: 0 });
+  });
+
+  test('custom compare function', () => {
+    const compare: (a: string, b: string) => number = (a: string, b: string) => b.localeCompare(a);
+    deepEqualInOrder(Sort.sortKeys({ c: 0, a: 0, b: 0 }, { compare }), { c: 0, b: 0, a: 0 });
+  });
+
+  test('deep option', () => {
+    deepEqualInOrder(Sort.sortKeys({ c: { c: 0, a: 0, b: 0 }, a: 0, b: 0 }, { deep: true }), {
+      a: 0,
+      b: 0,
+      c: { a: 0, b: 0, c: 0 }
+    });
+
+    expect(() => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const object: Partial<Record<string, any>> = { a: 0 };
+      object.circular = object;
+      Sort.sortKeys(object, { deep: true });
+    }).not.toThrow();
```

Comment:
If anything throws, the test will fail, so these assertions are unnecessary.

Reply:
Removed